src/module/order/application/repository/order.repository.interface.ts

```diff
@@ -1,11 +1,13 @@
+import { DeleteResult } from 'typeorm';
 import { Order } from '../../domain/order.entity';
 
 export const ORDER_REPOSITORY = 'ORDER_REPOSITORY';
 
 export interface IOrderRepository {
-  delete(orderId: number): Promise<void>;
+  delete(orderId: number): Promise<DeleteResult>;
```

Comment:
Typeorm types should not leak to the interface or the service. 
instead of returning DeleteResult, work the data to something like: Promise<Order>, void should be ok too.

Reply:
Done, now this return a Primise<void>